Update database-per-tenant.md (#2419)
neondatabase/website: content/use-cases/database-per-tenant.md

}}
/>

## Database-per-user in Neon

Neon is Postgres with serverless architecture. With rapid provisioning, scale-to-zero, and robust API support, you can scale database-per-user architectures without management overhead or big budgets. Just create **one project per customer** via the Neon API.

### One project per customer

A Neon project is the logical equivalent of an "instance" but without the management heaviness:

> - By creating one project per customer, each customers' data will be completely isolated.

Reviewer: ```suggestion
- By creating one project per customer, each customer's data will be completely isolated.
```

- You'll be able to run independent PITRs without affecting your entire fleet.

> - You can create diffeent projects in different regions to match your customers' location.

Reviewer: ```suggestion
- You can create different projects in different regions to match your customers' locations.
```

Management is simplified vs other Postgres services because,

- There’s no need to provision infrastructure in advance.

> - You can scale your architecture progressively, from a few tenants to hundreds of thousands, without breaking the bank - our pricing plans include a generous number of projects within the monthly fee.

Reviewer: ```suggestion
- You can scale your architecture progressively, from a few tenants to hundreds of thousands, without breaking the bank — our pricing plans include a generous number of projects within the monthly fee.
```

- Load your testing data to the main branch. This main branch acts as the primary source for all dev/test environments (they can be hundreds).
- To instantly create ephemeral environments, derive child branches from the main branch. These branches are fully isolated resource-wise and already include an up-to-date copy of the testing dataset. They can then be synced with the main branch with just one click.
- Once the work is complete, ephemeral dev/test environments (child branches) can be deleted automatically via your CI/CD.

## Neon for B2B SaaS: Data isolation with easy scalability

If you’re building a B2B SaaS platform, a database-per-tenant design can simplify your architecture while preserving scalability. With Neon, when you place its tenant on its own project, you offer complete data privacy to your customers via instance-level isolation. This approach also makes it easy to comply with data regulations across different regions, as projects can be created in specific locations to meet local requirements.

> Each tenant can be scaled independently, optimizing both performance and costs while reducing operational risk. And in the event of an issue / a customer request, you can [run point-in-time recovery (PITR) instantaneously for a specific tenant, without impacting the rest of the fleet](https://neon.tech/docs/guides/branch-restore).

Reviewer: ```suggestion
Each tenant can be scaled independently, optimizing both performance and costs while reducing operational risk. And in the event of an issue or a customer request, you can [run point-in-time recovery (PITR) instantaneously for a specific tenant, without impacting the rest of the fleet](https://neon.tech/docs/guides/branch-restore).
```

## Neon for dev platforms: Join Vercel, Replit, Koyeb, and others

> If you’re instead building a developer platform including a backend, or an [AI Agent](https://neon.tech/use-cases/ai-agents), you can start offering Neon databases to your users by becoming a [Partner](https://neon.tech/partners). Neon is a cost-effective solution to at the same time support your hobby plan and your Enterprise customers. Companies like [Vercel](https://neon.tech/blog/neon-postgres-on-vercel), [Replit](https://neon.tech/blog/neon-replit-integration), and [Koyeb](https://www.koyeb.com/blog/serverless-postgres-public-preview) are already using Neon to offer Postgres to their end-users.

Reviewer: ```suggestion
If you’re instead building a developer platform including a backend, or an [AI Agent](https://neon.tech/use-cases/ai-agents), you can start offering Neon databases to your users by becoming a [Partner](https://neon.tech/partners). Neon is a cost-effective solution that can support your hobby plan and Enterprise customers at the same time. Companies like [Vercel](https://neon.tech/blog/neon-postgres-on-vercel), [Replit](https://neon.tech/blog/neon-replit-integration), and [Koyeb](https://www.koyeb.com/blog/serverless-postgres-public-preview) are already using Neon to offer Postgres to their end-users.
```
